src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4499,6 +4519,7 @@ private TypeSymbol VisitArrayInitialization(TypeSymbol type, BoundArrayInitializ
                         var expression = GetConversionIfApplicable(expressions[i], expressionNoConversion);
                         expressionTypes[i] = VisitConversion(expression, expressionNoConversion, conversions[i], inferredType, expressionTypes[i], checkConversion: true,
                             fromExplicitCast: false, useLegacyWarnings: false, AssignmentKind.Assignment, reportRemainingWarnings: true, reportTopLevelWarnings: false);
+                        Unsplit();
```

Comment:
>Unsplit(); [](http://example.com/codeflow?start=24&length=10)

Are there tests that demonstrate the effect of this change?

Reply:
`NotNullWhen_UserDefinedConversion_Completeness` hits the new assertion without this because `VisitConversion` above leaves us in a split state in that scenario, which results in a split state at the end of the array initialization.